src/dune_rules/pkg_rules.ml

```diff
@@ -1198,16 +1205,37 @@ end = struct
           let doc =
             Path.outside_build_dir @@ Path.Outside_build_dir.relative prefix "doc"
           in
+          let* build_command =
+            match build_command with
+            | None | Some Dune -> Memo.return build_command
+            | Some (Action action) ->
+              let+ installed = Fs_memo.dir_exists prefix in
+              let action =
+                if installed
+                then
+                  (* If the toolchain is already installed, just create an empty
+                     config.cache file so other packages see that the compiler package
+                     is installed. *)
+                  Dune_lang.Action.Run
```

Comment:
Any reason why you moved this from `Pkg_toolchain`? I'd really prefer to keep this feature as isolated as possible. I realize, we can't fully accomplish that, but we should at least try.

Reply:
Not a strong reason, no. I'll move it back.